components/file_upload/file_upload.jsx

```diff
@@ -650,10 +656,11 @@ export default class FileUpload extends PureComponent {
                             openLeft={true}
                             openUp={true}
                             ariaLabel={formatMessage({id: 'file_upload.menuAriaLabel', defaultMessage: 'Upload type selector'})}
+                            customStyles={customStyles}
```

Comment:
Instead of using inline styles, could we instead pass in a class name for the menu to use? Passing an object literal is going to cause the Menu to re-render more than it needs to.

Reply:
Yep, that makes sense.

Reply:
We can define the styles object 
1. At the start of file 
2. in the constructor 

so we don't create new objects every-time to prevent re-renders. i am 0/5 on either of the above choices i am suggesting as this is one small object. 

While we are at this i have couple more thoughts on this: 
The difference between the two approaches is that if we define at the start of the file the browser wont be garbage collecting these and if in constructor we would have to create this object eveytime we open this dropdown.

So if we are creating hundreds of style objects for a menu then the tradeof would memory vs speed.  approach 1 is having to retain more objects in memory and 2 leads to time it takes for creating the new objects every-time a new menu opens. 
One more difference would be based on multiple menus at the same time as approach 1 will not create new ones but approach 2 needs to create new ones.

Reply:
@bradjcoughlin one minor change request here. it is preferred not to create objects we pass to react components in render method. Other than this other changes LGTM.